route/build/src/mpi_process.f90

```diff
@@ -1124,6 +1139,36 @@ subroutine mpi_restart(pid,           & ! input: proc id
     end do
   endif ! (onRoute(impulseResponseFunc))
 
+  if (tracer) then
+    ! tracer mass  communication
+    if (masterproc) then
+      do iSeg = 1, nRch
+        vol_global_tmp(iSeg) = RCHFLX(iens,iSeg)%ROUTE(idxDW)%reach_solute_mass(1)
+      enddo
+    else
+      vol_global_tmp(:) = realMissing
+    endif
+    call mpi_comm_single_flux(pid, nNodes,                              &
+                              vol_global_tmp,                           &
+                              vol_local,                                &
+                              rch_per_proc(root:nNodes-1),              &
+                              ixRch_order(rch_per_proc(root-1)+1:nRch), &
+                              arth(1,1,rch_per_proc(pid)),              &
```

Comment:
Is this passing a single element? Or just the first index that will be interpreted as an array?

If the later, you probably want to look into array segment syntax, and think about this carefully.

Reply:
`arth` is a function that returns incremental array from 1 to number of reach (==rch_per_proc(pid)).